foreshadow/console.py

```diff
@@ -1,269 +1,266 @@
+
+
+def execute_model(fs, X_train, y_train, X_test, y_test):
+    """Execute the model produced by `generate_model()`.
+
+    Also, exports the data to json and returns the exported json object
+    containing the results and the serialized Foreshadow object. Also, prints
+    simple model accuracy metrics.
+
+    Args:
+        fs (foreshadow.Foreshadow): An unfit foreshadow object.
+        X_train (:obj:`DataFrame <pandas.DataFrame>`): The X train data.
+        X_test (:obj:`DataFrame <pandas.DataFrame>`): The X test data.
+        y_train (:obj:`DataFrame <pandas.DataFrame>`): The y train data.
+        y_test (:obj:`DataFrame <pandas.DataFrame>`): The y test data.
+
+    Returns:
+        dict: A dictionary with the following keys `X_Model`, `X_Summary`, \
+            `y_model`, and `y_summary` which each represent the serialized \
+            and summarized forms of each of those steps.
+
+    """
+    print("Fitting final model...")
+    fs.fit(X_train, y_train)
+
+    print("Scoring final model...")
+    score = fs.score(X_test, y_test)
+
+    print("Final Results: ")
+    print(score)
+
+    fs.to_json("foreshadow.json", deep=True)
+    print(
+        "Serialized foreshadow pipeline has been saved to foreshadow.json. "
+        "Refer to docs to read and process."
+    )
+    # TODO serialize the foreshadow object and summarize the X and y stats.
+    # Store final results
+    # all_results = {
+    #     "X_Model": fs.X_preparer.serialize(),
+    #     # "X_Summary": fs.X_preparer.summarize(X_train),
+    #     "y_Model": fs.y_preparer.serialize(),
+    #     # "y_summary": fs.y_preparer.summarize(y_train),
+    # }
+    # return all_results
+
+
+def cmd():  # pragma: no cover
+    """Entry point to foreshadow via console command.
+
+    Uncovered as this function only serves to be executed manually.
+    """
+    model = generate_model(sys.argv[1:])
+    execute_model(*model)
+
+
+def get_method(arg, y_train):
+    """Determine what estimator to use.
+
+    Uses set of X data and a passed argument referencing an
+    `BaseException <sklearn.base.BaseEstimator>` class.
+
+    Args:
+        arg (str): model name
+        y_train (:obj:`DataFrame <pandas.DataFrame>`): The response variable
+            data.
+
+    Returns:
+        Estimator
+
+    Raises:
+        ValueError: if invalid method is chosen
+
+    """
+    if arg is not None:
+        try:
+            mod = __import__(
+                "sklearn.linear_model", globals(), locals(), ["object"], 0
+            )
+            cls = getattr(mod, arg)
+            return cls()
+        except Exception:
+            raise ValueError(
+                "Invalid method. {} is not a valid "
+                "estimator from sklearn.linear_model".format(arg)
+            )
+    else:
+        return (
+            LinearRegression()
+            if determine_problem_type(y_train) == "regression"
+            else LogisticRegression()
+        )
+
+
```

Comment:
This is just an uncomment, right?

Reply:
Mostly, yes. The only additional change is that I added a line of code to serialize the whole foreshadow object instead of the X_preprocessor and y_preprocessor in the old code.